### DISCUSSION TITLE
New Bulit-in Service: Register unique identity to global.

### DISCUSSION THREAD

**Discussion by sunhuachuang** [CympleTech/esse#3] [Ideas]
## What ?
A service for search friend in the network use unique identify name/username.

## Why ?
Now, when want to find a friend, need input the friend's did and network address. But it is not easy to remember and input. so, Maybe we need a unique identify service to search friend by name or custom username.
```

----------    register    --------------------------------------------------
--      --   ---------->  --                                              --
--      --     update     --                                              --
--      --                --  ---------------       --------------------  --
-- ESSE --                --  - TDN gateway -  -->  - Back-end service -  --
--      --     search     --  ---------------       --------------------  --
--      --   --------->   --                                              --
--      --   <--------    --             Identity service                 --
----------    UserInfo    --------------------------------------------------

```
## How ? 
1. we need a gateway that Link web2.0 (Traditional back-end website) and TDN applications. [Discuss Here ](https://github.com/cypherlink/TDN/discussions/8)
2. ESSE need support connect to other TDN-based service (not only other ESSE node)
3. This service is a protocol, and can be changed by the user.
4. This service register store user's nickname, did, peer address, avatar, bio to global database for search.
5. user can delete it anytime.
6. Add web service to handle search and show (can choose show direct or show with some key)

**Comment by tiannian** [comment]
I think ESSE can use different server as identity provider, use `username@provider.com` to find user.
It can also use blockchain as provider to regist username.

**Reply by sunhuachuang** [reply]
Yes, good idea, user can add (choose) different provider, and also can upload info to different provider. (the provider maybe permissionless or permissioned).

**Reply by tiannian** [reply]
I think is a best compatible plan for ESSE
